test/built-ins/RegExp/named-groups/duplicate-names-matchall.js

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 Igalia S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: String.prototype.search behavior with duplicate named capture groups
+esid: prod-GroupSpecifier
+includes: [compareArray.js]
+features: [regexp-duplicate-named-groups]
+---*/
+
+function matchesIterator(iterator, expected) {
+  const actual = [...iterator];
+  assert.sameValue(actual.length, expected.length, "lengths should be equal");
+  for (let i = 0; i < expected.length; ++i) {
+    assert.compareArray(actual[i], expected[i]);
+  }
+}
```

Comment:
You could consider using [`assert.compareIterator`](https://github.com/tc39/test262/blob/main/harness/compareIterator.js) to reduce boilerplate, but not a blocker.